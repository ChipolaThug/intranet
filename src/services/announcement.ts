export class Announcement {
    id = 0

    created = new Date()

    type = ""

    title = ""

    url = ""
}

export const translationAnnouncement: { [k in keyof Announcement]: string } = {
    id: "id",
    created: "creation",
    type: "type",
    title: "titre",
    url: "url",
}

export const elementName = "Announcement"

export type AnnouncementWithoutId = Omit<Announcement, "id">
