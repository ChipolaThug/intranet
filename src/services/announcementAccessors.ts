import ServiceAccessors from "./accessors"
import { elementName, Announcement, AnnouncementWithoutId } from "./announcement"

const serviceAccessors = new ServiceAccessors<AnnouncementWithoutId, Announcement>(elementName)

// export const announcementGet = serviceAccessors.get()
// export const announcementAdd = serviceAccessors.add()
// export const announcementSet = serviceAccessors.set()

export const announcementListGet = serviceAccessors.securedListGet()
