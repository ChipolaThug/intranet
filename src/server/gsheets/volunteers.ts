import getExpressAccessors from "./expressAccessors"
import { sheetNames } from "./accessors"
import { Volunteer, VolunteerWithoutId, translationMember } from "../../services/volunteers"

const { listGetRequest, getRequest, setRequest, addRequest } = getExpressAccessors<
    VolunteerWithoutId,
    Volunteer
>(sheetNames.Volunteers, new Volunteer(), translationMember)

export const volunteerListGet = listGetRequest()

export const volunteerGet = getRequest()

export const volunteerAdd = addRequest()

export const volunteerSet = setRequest()
