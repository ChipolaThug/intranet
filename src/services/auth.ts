import { AxiosRequestConfig } from "axios"
import Cookies from "js-cookie"

import { VolunteerLogin } from "./volunteers"

export const axiosConfig: AxiosRequestConfig = {
    headers: {},
}

export function setJWT(token: string, id: number): void {
    axiosConfig.headers.Authorization = `Bearer ${token}`
    Cookies.set("jwt", token, { expires: 3650 })
    Cookies.set("id", `${id}`, { expires: 3650 })
}

export function unsetJWT(): void {
    delete axiosConfig.headers.Authorization

    Cookies.remove("jwt")
    Cookies.remove("id")
}

export function getCookieJWT(cookie = ""): VolunteerLogin {
    const cookies = cookie
        .split(";")
        .reduce((res: { [cookieName: string]: string }, el: string) => {
            const [k, v] = el.split("=")
            res[k.trim()] = v
            return res
        }, {})
    return { jwt: cookies.jwt, id: +cookies.id }
}
