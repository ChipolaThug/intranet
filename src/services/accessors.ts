import axios from "axios"

import config from "../config"
import { axiosConfig } from "./auth"

export type ElementWithId = unknown & { id: number }

export type ElementTranslation<Element> = { [k in keyof Element]: string }

export default function getServiceAccessors<
    // eslint-disable-next-line @typescript-eslint/ban-types
    ElementNoId extends object,
    Element extends ElementNoId & ElementWithId
>(elementName: string): any {
    function get(): (id: number) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (id: number): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.get(`${config.API_URL}/${elementName}Get`, {
                    ...axiosConfig,
                    params: { id },
                })
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function listGet(): () => Promise<{
        data?: Element[]
        error?: Error
    }> {
        interface ElementListGetResponse {
            data?: Element[]
            error?: Error
        }
        return async (): Promise<ElementListGetResponse> => {
            try {
                const { data } = await axios.get(
                    `${config.API_URL}/${elementName}ListGet`,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    // eslint-disable-next-line @typescript-eslint/ban-types
    function add(): (volunteerWithoutId: ElementNoId) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (volunteerWithoutId: ElementNoId): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}Add`,
                    volunteerWithoutId,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function set(): (volunteer: Element) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (volunteer: Element): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}Set`,
                    volunteer,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function countGet(): () => Promise<{
        data?: number
        error?: Error
    }> {
        interface ElementCountGetResponse {
            data?: number
            error?: Error
        }
        return async (): Promise<ElementCountGetResponse> => {
            try {
                const { data } = await axios.get(
                    `${config.API_URL}/${elementName}CountGet`,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function customPost(apiName: string): (params: any) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (params: any): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}${apiName}`,
                    params,
                    axiosConfig
                )
                if (data.error) {
                    throw Error(data.error)
                }
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    return { listGet, get, set, add, countGet, customPost }
}
