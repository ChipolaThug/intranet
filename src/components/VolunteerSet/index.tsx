import React, { useState, memo } from "react"
import { toast } from "react-toastify"

import { AppDispatch } from "../../store"

import { fetchVolunteerSet } from "../../store/volunteerSet"
import { Volunteer } from "../../services/volunteers"
import styles from "./styles.module.scss"

interface Props {
    dispatch: AppDispatch
    volunteer: Volunteer
}

const VolunteerSet = ({ dispatch, volunteer }: Props) => {
    const [firstname, setFirstname] = useState(volunteer.firstname)
    const [lastname, setName] = useState(volunteer.lastname)
    const [adult, setAdult] = useState(volunteer.adult)

    const onFirstnameChanged = (e: React.ChangeEvent<HTMLInputElement>) =>
        setFirstname(e.target.value)
    const onNameChanged = (e: React.ChangeEvent<HTMLInputElement>) => setName(e.target.value)
    const onAdultChanged = (e: React.ChangeEvent<HTMLInputElement>) => setAdult(+e.target.value)

    const onSavePostClicked = () => {
        if (firstname && lastname) {
            dispatch(
                fetchVolunteerSet({
                    ...volunteer,
                    firstname,
                    lastname,
                    adult,
                })
            )
        } else {
            toast.warning("Il faut au moins préciser un prenom et un nom", {
                position: "top-center",
                autoClose: 6000,
                hideProgressBar: true,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
            })
        }
    }
    return (
        <section className={styles.VolunteerList}>
            <h2>Modifier un volunteer</h2>
            <form>
                <label htmlFor="postFirstname">
                    Prénom:
                    <input
                        type="text"
                        id="postFirstname"
                        name="postFirstname"
                        value={firstname}
                        onChange={onFirstnameChanged}
                    />
                </label>
                <label htmlFor="postName">
                    Nom:
                    <input
                        type="text"
                        id="postName"
                        name="postName"
                        value={lastname}
                        onChange={onNameChanged}
                    />
                </label>
                <label htmlFor="postAdult">
                    Majeur:
                    <input
                        type="text"
                        id="postAdult"
                        name="postAdult"
                        value={adult}
                        onChange={onAdultChanged}
                    />
                </label>
                <button type="button" onClick={onSavePostClicked}>
                    Save changes
                </button>
            </form>
        </section>
    )
}
export default memo(VolunteerSet)
