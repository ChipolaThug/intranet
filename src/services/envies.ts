import { get, listGet, add, set } from "./accessors"

export class Envie {
    id = 0

    domaine = ""

    envies = ""

    precisions = ""

    equipes: string[] = []

    dateAjout = ""
}

export type EnvieWithoutId = Omit<Envie, "id">

export const envieGet = get<Envie>("Envie")

export const envieListGet = listGet<Envie>("Envie")

export const envieAdd = add<EnvieWithoutId, Envie>("Envie")

export const envieSet = set<Envie>("Envie")
