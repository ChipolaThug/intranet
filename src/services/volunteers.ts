import { get, listGet, add, set } from "./accessors"

export class Volunteer {
    id = 0

    lastname = ""

    firstname = ""

    email = ""

    mobile = ""

    photo = ""

    food = ""

    adult = 1

    privileges = 0

    active = 0

    comment = ""

    timestamp = ""

    password = ""
}

export const translationMember: { [k in keyof Volunteer]: string } = {
    id: "id",
    lastname: "nom",
    firstname: "prenom",
    email: "mail",
    mobile: "telephone",
    photo: "photo",
    food: "alimentation",
    adult: "majeur",
    privileges: "privilege",
    active: "actif",
    comment: "commentaire",
    timestamp: "horodatage",
    password: "passe",
}

const elementName = "Volunteer"

export const emailRegexp =
    /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i
export const passwordMinLength = 4

export interface MemberLogin {
    volunteer?: {
        firstname: string
    }
    jwt?: string
    error?: string
}

export type VolunteerWithoutId = Omit<Volunteer, "id">

export const volunteerGet = get<Volunteer>(elementName, translationMember)

export const volunteerListGet = listGet<Volunteer>(elementName, translationMember)

export const volunteerAdd = add<VolunteerWithoutId, Volunteer>(elementName, translationMember)

export const volunteerSet = set<Volunteer>(elementName, translationMember)
