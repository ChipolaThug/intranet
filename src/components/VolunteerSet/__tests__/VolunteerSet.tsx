/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { volunteerExample } from "../../../services/volunteers"

import VolunteerSet from "../index"

describe("<SetVolunteer />", () => {
    it("renders", () => {
        const dispatch = jest.fn()
        const tree = render(
            <MemoryRouter>
                <VolunteerSet dispatch={dispatch} volunteer={volunteerExample} />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
