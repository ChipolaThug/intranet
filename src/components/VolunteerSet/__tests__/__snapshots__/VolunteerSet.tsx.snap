// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetVolunteer /> renders 1`] = `
<section
  class="VolunteerList"
>
  <h2>
    Modifier un volunteer
  </h2>
  <form>
    <label
      for="postFirstname"
    >
      Prénom:
      <input
        id="postFirstname"
        name="postFirstname"
        type="text"
        value="Aupeix"
      />
    </label>
    <label
      for="postName"
    >
      Nom:
      <input
        id="postName"
        name="postName"
        type="text"
        value="Amélie"
      />
    </label>
    <label
      for="postAdult"
    >
      Majeur:
      <input
        id="postAdult"
        name="postAdult"
        type="text"
        value="1"
      />
    </label>
    <button
      type="button"
    >
      Save changes
    </button>
  </form>
</section>
`;
